cli/ops/net_unix.rs

```diff
@@ -30,25 +33,39 @@ pub struct UnixListenArgs {
 }
 
 pub(crate) async fn accept_unix(
-  state: Rc<State>,
+  state: Rc<RefCell<OpState>>,
   args: AcceptArgs,
   _bufs: BufVec,
 ) -> Result<Value, ErrBox> {
   let rid = args.rid as u32;
 
-  let mut resource_table_ = state.resource_table.borrow_mut();
-  let listener_resource = {
-    resource_table_
+  let accept_fut = poll_fn(|cx| {
+    let mut state = state.borrow_mut();
+    let listener_resource = state
+      .resource_table
       .get_mut::<UnixListenerResource>(rid)
-      .ok_or_else(|| ErrBox::bad_resource("Listener has been closed"))?
-  };
-  let (unix_stream, _socket_addr) = listener_resource.listener.accept().await?;
-  drop(resource_table_);
+      .ok_or_else(|| ErrBox::bad_resource("Listener has been closed"))?;
+    let listener = &mut listener_resource.listener;
+    use futures::StreamExt;
+    match listener.poll_next_unpin(cx) {
+      Poll::Ready(Some(stream)) => {
+        //listener_resource.untrack_task();
```

Comment:
What's going on here?

Reply:
Working around a broken test... but maybe it can be undone

Reply:
Working around a broken test... but maybe it can be undone